core/trino-main/src/main/java/io/trino/operator/aggregation/DecimalAverageAggregation.java

```diff
@@ -161,27 +161,34 @@ public static void inputLongDecimal(Type type, LongDecimalWithOverflowAndLongSta
 
     private static void accumulateValueInState(Slice unscaledValue, LongDecimalWithOverflowAndLongState state)
     {
-        accumulateAndUpdateOverflow(unscaledValue, state);
-        state.setLong(state.getLong() + 1);
-    }
-
-    private static void initializeIfNeeded(LongDecimalWithOverflowAndLongState state)
-    {
-        if (state.getLongDecimal() == null) {
-            state.setLongDecimal(UnscaledDecimal128Arithmetic.unscaledDecimal());
+        Slice sum = state.getLongDecimal();
+        if (sum == null) {
+            sum = UnscaledDecimal128Arithmetic.unscaledDecimal();
+            state.setLongDecimal(sum);
         }
+        long overflow = UnscaledDecimal128Arithmetic.addWithOverflow(sum, unscaledValue, sum);
+        if (overflow != 0) {
+            state.setOverflow(state.getOverflow() + overflow);
```

Comment:
Seems like `state.addOverflow(overflow)` would suffice here.

Reply:
oops... yeah, missed this one